### ISSUE TITLE
Additional manual scrape configs

### ISSUE THREAD

**Issue by corvus-ch** [vfarcic/docker-flow-monitor#4]
I have setup a swarm cluster with docker flow monitor for monitoring. All is working fine so far. No I have the use case where I have to define some scrape configs manually. 

Why do I need to write this scrape config manually? I am using blackbox_exporter for probing some external service (outside of my control) and for that I need a bunch of relabel statements.

As far as I can see, this is not supported. Since I can nighter set relabelling thought docker labels nor I can define some custom scrape configs. How should I approach this use case?

Thanks in advance for help.

**Comment by vfarcic** [comment]
PR #3 will add the feature you need. @albertogviana is working on it. It should be available soon.

**Comment by corvus-ch** [comment]
Thanks for the fast reply.

The solution in PR #3 will solve my problem only partially as it does not address my need for relabelling. Please have a look at the following Prometheus config snipped.

```
scrape_configs:
  - job_name: example
    dns_sd_configs:
      - names: ["example.com"]
        type: A
        port: 443
    metrics_path: /probe
    params:
      module: [example]
    relabel_configs:
      - source_labels: [__address__]
        regex: (.*)
        target_label: __param_target
        replacement: https://${1}/v1/rest/resource
      - source_labels: [__address__]
        target_label: instance
      - target_label: __address__
        replacement: blackbox_exporter:9115
```

**Comment by vfarcic** [comment]
@corvus-ch You're right. That PR focuses on a specific use case. We should make it more generic instead and allow any scrape config parameter.

One solution could be similar to [global-variables](http://monitor.dockerflow.com/config/#global-variables). We could say that any env. var. prefixed with `SCRAPE_` should be converted into a scrape config entry. Those variables would need to be suffixed with an index so that multiple configs could be specified. However, looking at your example config, that might lead to unmaintainable env. vars.

Maybe it would be better to have a system that would allow additional scrape configs to be defined as files. They could be, for example, Docker secrets.

What do you think? Would that suit your use case?

**Comment by corvus-ch** [comment]
I agree, following the example of the global variables will most probably end up in a mess. Embedding files is actually what I have searched for. If this would be possible, they could be added as docker config  and or secret. For that reason I suggest to allow to add any number of files.

**Comment by vfarcic** [comment]
It's available as the [release 17.08.07-6](https://github.com/vfarcic/docker-flow-monitor/releases/tag/17.08.07-6). Can you please try it out and let me know whether it works as expected?

**Comment by corvus-ch** [comment]
Thanks that worked as expected.

Is it by intention that only secrets are supported. There is also the concept of configs (https://docs.docker.com/engine/swarm/configs/). Technically they work (more or less) the same and I can easily use secrets. Semantically I would prefer to use configs though.

**Comment by vfarcic** [comment]
Scrape snippets can now be loaded from any directory. It is available from the [release 17.08.08-9](https://github.com/vfarcic/docker-flow-monitor/releases/tag/17.08.08-9). Please let me know if you have any other suggestion how we can improve this or any other aspect of the project.

**Comment by corvus-ch** [comment]
Thanks a lot. That worked nice and smooth.

I really appreciate how fast issues are taken care of.
